docs/doxygen/overviews/high_dpi.md

```diff
@@ -69,55 +69,194 @@ pixels**, abbreviated as "DIP", that are always of the same size on all
-drawing graphics elements to ensure that the best possible resolution is used.
-For example, all operations on wxGLCanvas use physical pixels.
+Physical pixels are never scaled, on any platform, and are used for the real
+bitmap sizes. They are also used for drawing operations on wxGLCanvas, which
+is _different_ from wxDC and wxGraphicsContext that use logical pixels.
 
-To convert between logical and physical pixels, you can use
-wxWindow::GetContentScaleFactor(): this is a value greater than or equal to 1,
-so a value in logical pixels needs to be multiplied by it in order to obtain
-the value in physical pixels.
+Under MSW physical pixels are same as logical ones, but when writing portable
+code you need to convert between logical and physical pixels using
+wxWindow::GetContentScaleFactor(): this function returns a value greater than
+or equal to 1 (and always just 1 under MSW), so a value in logical pixels
+needs to be multiplied by it in order to obtain the value in physical pixels.
 
 For example, in a wxGLCanvas created with the size of 100 (logical) pixels, the
-rightmost physical pixel coordinate will be `100*GetContentScaleFactor()`.
+rightmost physical pixel coordinate will be `100*GetContentScaleFactor()`
+under all platforms.
+
+You can convert from DIPs to physical pixels by converting DIPs to the logical
+pixels first, but you can also do it directly, by using
+wxWindow::GetDPIScaleFactor(). This function can return a value different from
+1 even under MSW, i.e. it returns DPI scaling for physical display pixels.
+
+Summary of Different Pixel Kinds    {#high_dpi_pixel_conversions}
+--------------------------------
+
+Under MSW, logical pixels are always the same as physical pixels, but are
+different from DIPs, while under all the other platforms with DPI scaling
+support (currently only GTK 3 and macOS), logical pixels are the same as DIP,
+but different from physical pixels.
+
+However under all platforms the following functions can be used to convert
+between different kinds of pixels:
+
+* From DIP to logical pixels: use wxWindow::FromDIP() or wxWindow::ToDIP().
+* Logical pixels and physical pixels: multiply of divide by wxWindow::GetContentScaleFactor().
+* From DIP to physical pixels: multiply of divide by wxWindow::GetDPIScaleFactor().
 
+Or, in the diagram form:
 
-High-Resolution Images and Artwork
+@dot
+digraph Pixels
+{
+    node [shape = hexagon, style = filled];
+
+    LP  [fillcolor = lightblue, label = "Logical\nPixels"];
+    DIP [fillcolor = yellow, label = "DI\nPixels"];
+    PP  [fillcolor = green, label = "Physical\nPixels"];
+
+    LP  -> PP [dir = both, minlen = 3, headlabel = "mul by GetContentScaleFactor()", taillabel = "div by GetContentScaleFactor()"];
+    LP -> DIP [dir = both, weight = 2, minlen = 3, headlabel = "ToDIP()", taillabel = "FromDIP()"];
+    DIP -> PP [dir = both, minlen = 15, headlabel = "mul by GetDPIScaleFactor()", taillabel = "div by GetDPIScaleFactor()", constraint = false] ;
+}
```

Comment:
how would this look like on your machine:

```
digraph Pixels
{
    node [shape = hexagon, style = filled, fontname = Helvetica];

    LP  [fillcolor = lightblue, label = "Logical\nPixels"];
    DIP [fillcolor = yellow, label = "DI\nPixels"];
    PP  [fillcolor = green, label = "Physical\nPixels"];

    LP  -> PP [fontname = Helvetica, labeldistance = 5, labelangle = 30, dir = both, weight = 2, minlen = 3, label = "GetContentScaleFactor()", headlabel = "multiply by", taillabel = "divide by"];
    LP -> DIP [fontname = Helvetica, labeldistance = 6, dir = both, weight = 2, minlen = 3, headlabel = "ToDIP()", taillabel = "FromDIP()"];
    DIP -> PP [fontname = Helvetica, dir = both, minlen = 10, label = "GetDPIScaleFactor()" headlabel = "multiply by", taillabel = "divide by", constraint = false] ;
}
```

Reply:
Much better, thanks!

BTW, I was using [dot visual editor](http://magjac.com/graphviz-visual-editor/) for testing this, I didn't know about it before, but it's really useful.

Reply:
> Much better, thanks!
> 
> BTW, I was using [dot visual editor](http://magjac.com/graphviz-visual-editor/) for testing this, I didn't know about it before, but it's really useful.

Thanks, yes, didn't know it either, always great to have something browser-based. I've found the VS-Code extension Graphviz (dot) quite useful as well.